redisvl/extensions/llmcache/semantic.py

```diff
@@ -16,6 +18,9 @@ class SemanticCache(BaseLLMCache):
     entry_id_field_name: str = "id"
     prompt_field_name: str = "prompt"
     vector_field_name: str = "prompt_vector"
+    inserted_at_field_name: str = "inserted_at"
+    updated_at_field_name: str = "updated_at"
+    tag_field_name: str = "scope_tag"
```

Comment:
Just want to understand this data model a bit more. As it stands here, it looks like there would be 1 field for scoping (`scope_tag`) and filtering. What if I have both year, location, and user ID data that I'd like to filter my cache check on?

I think we might need to generalize a bit more and support a set of optional/customizable filterable fields (providable to the class on init??)

So if staying with Hash, the schema could be something like:
- `id`
- `prompt`
- `prompt_vector`
- `response`
- `inserted_at`
- `updated_at`
- `metadata`
- `scope_tag`???
- <all other fields as customizable filter fields included in the cache schema>?

Just thinking out loud!